src/init.cpp

```diff
@@ -170,12 +171,24 @@ static fs::path GetPidFile(const ArgsManager& args)
 #else
         tfm::format(file, "%d\n", getpid());
 #endif
+        g_generated_pid = true;
         return true;
     } else {
         return InitError(strprintf(_("Unable to create the PID file '%s': %s"), fs::PathToString(GetPidFile(args)), SysErrorString(errno)));
     }
 }
 
+static void RemovePidFile(const ArgsManager& args)
```

Comment:
nit: quick docstring would be nice

```
/**
 * Remove PID file, but only if it was also generated by this process to avoid problems when
 * multiple bitcoind processes are started on the same datadir.
 */
```

<details>
<summary>git diff on 82f18e907a</summary>

```diff
diff --git a/src/init.cpp b/src/init.cpp
index fced24a6ff..7a87467252 100644
--- a/src/init.cpp
+++ b/src/init.cpp
@@ -177,7 +177,10 @@ static fs::path GetPidFile(const ArgsManager& args)
         return InitError(strprintf(_("Unable to create the PID file '%s': %s"), fs::PathToString(GetPidFile(args)), SysErrorString(errno)));
     }
 }
-
+/**
+ * Remove PID file, but only if it was also generated by this process to avoid problems when
+ * multiple bitcoind processes are started on the same datadir.
+ */
 static void RemovePidFile(const ArgsManager& args)
 {
     if (!g_generated_pid) return;

```
</details>